dasgo/resources.go

```diff
@@ -444,53 +459,55 @@ const (
 // Optional Audit Entry Info
 // https://discord.com/developers/docs/resources/audit-log#audit-log-entry-object-optional-audit-entry-info
 type AuditLogOptions struct {
-	ApplicationID    Snowflake `json:"application_id,omitempty"`
-	ChannelID        Snowflake `json:"channel_id,omitempty"`
-	Count            string    `json:"count,omitempty"`
-	DeleteMemberDays string    `json:"delete_member_days,omitempty"`
-	ID               Snowflake `json:"id,omitempty"`
-	MembersRemoved   string    `json:"members_removed,omitempty"`
-	MessageID        Snowflake `json:"message_id,omitempty"`
-	RoleName         string    `json:"role_name,omitempty"`
-	Type             string    `json:"type,omitempty"`
+	ApplicationID    Snowflake `json:"application_id"`
+	ChannelID        Snowflake `json:"channel_id"`
+	Count            string    `json:"count"`
+	DeleteMemberDays string    `json:"delete_member_days"`
+	ID               Snowflake `json:"id"`
+	MembersRemoved   string    `json:"members_removed"`
+	MessageID        Snowflake `json:"message_id"`
+	RoleName         string    `json:"role_name"`
+	Type             string    `json:"type"`
 }
 
 // Audit Log Change Object
 // https://discord.com/developers/docs/resources/audit-log#audit-log-change-object
 type AuditLogChange struct {
 	NewValue interface{} `json:"new_value,omitempty"`
 	OldValue interface{} `json:"old_value,omitempty"`
-	Key      string      `json:"key,omitempty"`
+	Key      string      `json:"key"`
 }
 
 // Audit Log Change Exceptions
 // https://discord.com/developers/docs/resources/audit-log#audit-log-change-object-audit-log-change-exceptions
 
+// TODO: not sure for last pin timestamp
```

Comment:
Timestamps are probably not valid as a zero value. Time can be considered a struct and be a non-pointer because the user can use its `isZero` command to determine whether it's a zero value.